api/app/lib/statistics.rb

```diff
@@ -0,0 +1,104 @@
+require "google/apis/analytics_v3"
+require "google/api_client/auth/key_utils"
+
+class Statistics
+  include ActiveModel::Validations
+  include ActiveModel::Conversion
+  extend ActiveModel::Naming
+
+  def id
+    "statistics-set"
+  end
+
+  def initialize(_range = nil)
+    @client = Google::Apis::AnalyticsV3::AnalyticsService.new
+
+    @client.authorization = Signet::OAuth2::Client.new(
+      token_credential_uri:  "https://accounts.google.com/o/oauth2/token",
+      audience:              "https://accounts.google.com/o/oauth2/token",
+      scope:                 "https://www.googleapis.com/auth/analytics.readonly",
+      issuer:                 ENV["GOOGLE_SERVICE_CLIENT_EMAIL"],
+      signing_key:            OpenSSL::PKey::RSA.new(ENV["GOOGLE_SERVICE_PRIVATE_KEY"])
+    ).tap(&:fetch_access_token!)
+
+    @range ||= Time.zone.today.at_beginning_of_week..Time.zone.today.at_end_of_week
+  end
+
+  def start_date
+    @range.first
+  end
+
+  def end_date
+    @range.last
+  end
+
+  def readers_this_week
+    get_reader_count(start_date, end_date)
+  end
+
+  def readers_last_week
+    get_reader_count(start_date - 7.days, end_date - 7.days)
+  end
+
+  def reader_increase
+    this_week = readers_this_week
+    last_week = readers_last_week
+
+    if last_week.zero?
+      result = this_week.positive? ? (this_week * 100).to_i : 0
+    else
+      diff = last_week - this_week
+      result = diff.zero? ? 0 : ((diff / last_week) * 100).to_i
+    end
+    result
+  end
+
+  def new_highlights_count
+    Annotation.where("created_at >= ? AND created_at <= ? AND format = ?",
+                     start_date,
+                     end_date,
+                     "highlight")
+              .count
+  end
+
+  def new_annotations_count
+    Annotation.where("created_at >= ? AND created_at <= ? AND format = ?",
+                     start_date,
+                     end_date,
+                     "annotation")
+              .count
+  end
+
+  # def new_comments_count
+  #   Annotation.where("created_at >= ? AND created_at <= ? AND format = ?",
+  #                    start_date,
+  #                    end_date,
+  #                    "comment")
+  #             .count
+  # end
+
+  def new_texts_count
+    Text.where("created_at >= ? AND created_at <= ?",
+               start_date,
+               end_date)
+        .count
+  end
+
+  def read_attribute_for_serialization(n)
+    send(n)
+  end
+
+  protected
+
+  def get_reader_count(start_date, end_date)
+    @client.get_ga_data(
+      ENV["GOOGLE_ANALYTICS_PROFILE_ID"], # analytics profile ID
```

Comment:
What does the profile ID look like? I wonder if it should be stored in the environment after all. How does one get one's profile ID from google?